File: miniapp/src/api/recommendation.ts
/**
 * 推荐内容 API
 */
import request from '@/utils/request'

const PREFIX = '/api/v1/recommendations'

/** 热门帖子推荐 */
export function getHotPosts(limit?: number) {
  return request({
    url: PREFIX + '/hot',
    method: 'GET',
    params: { limit: limit || 10 },
  })
}

/** 个性化推荐（需要登录） */
export function getPersonalizedFeed(limit?: number) {
  return request({
    url: PREFIX + '/feed',
    method: 'GET',
    params: { limit: limit || 20 },
  })
}

/** 推荐话题 */
export function getRecommendedTopics(limit?: number) {
  return request<{ items: TopicItem[] }>({
    url: PREFIX + '/topics',
    method: 'GET',
    params: { limit: limit || 10 },
  })
}

/** 推荐话题项 */
export interface TopicItem {
  id: string
  name: string
  description: string | null
  cover_image: string | null
  post_count: number
}

File: backend/app/api/v1/recommendation.py
"""
内容推荐接口 - 热门帖子、个性化推荐、话题推荐
"""
from fastapi import APIRouter, Depends, Query

from sqlalchemy.ext.asyncio import AsyncSession

from app.core.database import get_db
from app.core.deps import get_current_user
from app.models.user import User
from app.schemas.post import PostResponse
from app.services import recommendation_service

router = APIRouter(prefix="/recommendations", tags=["recommendations"])


@router.get("/hot", response_model=list[PostResponse])
async def get_hot_posts(
    limit: int = Query(10, ge=1, le=50, description="返回数量"),
    db: AsyncSession = Depends(get_db),
):
    """热门帖子推荐（基于点赞、评论、浏览热度）。"""
    posts = await recommendation_service.recommend_posts_hot(db, limit)
    return [PostResponse.model_validate(p) for p in posts]


@router.get("/feed", response_model=list[PostResponse])
async def get_personalized_feed(
    limit: int = Query(20, ge=1, le=100, description="返回数量"),
    current_user: User = Depends(get_current_user),
    db: AsyncSession = Depends(get_db),
):
    """个性化推荐（基于关注用户的协同过滤）。"""
    posts = await recommendation_service.recommend_posts_for_user(
        db, current_user.id, limit
    )
    return [PostResponse.model_validate(p) for p in posts]


@router.get("/topics")
async def get_recommended_topics(
    limit: int = Query(10, ge=1, le=50, description="返回数量"),
    db: AsyncSession = Depends(get_db),
):
    """推荐话题（按帖子数和活跃度）。"""
    topics = await recommendation_service.recommend_topics(db, limit)
    return {"items": topics}
